pagopa/io-app
# feat: [IOBP-634] Add pull to refresh in payments home screen

## Short description
This PR adds the pull-to-refresh feature in the payments home screen removing the previous refresh on focus showing the skeletons (now the skeletons will appear when it's the first time opening that screen)

## List of changes proposed in this pull request
- Replaced `useFocusEffect` -> `useOnFirstRender` to fetch the data to display in that screen only once.
- Added the refreshControl to the `IOScrollView` to force-refresh the data
- Added the data fetching based on the following occurrences:
    - When deleting a payment method
    - When adding a payment method
    - When completing a payment transaction

## How to test
- Open the payments home screen page, for the first time you open it you should be able to see the skeletons loading. Then, if you go into that screen nothing should happen until you pull to refresh the content where a refresh control spinner will appear on the top of the screen.
To test correctly, try to:
- Add a payment method
- Delete a payment method
- Completing a payment
All the above steps have to refresh the page without showing the skeletons.

## Preview
|Before|After|
|-|-|
|<video src="https://github.com/user-attachments/assets/a31f0dd9-109c-47af-be0c-efc5c4a94ffe" />|<video src="https://github.com/user-attachments/assets/2734a76d-4e12-4487-9279-da80b777cc41" />
